src/components/sectionList/modelValue/ModelValueRenderer.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import { ConstantValue } from './ConstantValue'
+import { DateValue } from './DateValue'
+import type { ValueDetails } from './ModelValue'
+import { PublicAccessValue, isSharing } from './PublicAccess'
+import { TextValue } from './TextValue'
+
+export const ModelValueRenderer = ({ value, schemaProperty }: ValueDetails) => {
+    if (schemaProperty.fieldName === 'sharing' && isSharing(value)) {
+        return <PublicAccessValue value={value} />
+    }
+
+    if (schemaProperty.propertyType === 'CONSTANT') {
```

Comment:
We could add type-guards for value here, instead of relying on `as`. But I think this should be safe-enough, and I couldn't decide on where the type-guard should "live"...

Reply:
I like the idea of type guards (just as you're already doing `isSharing` above), but don't think it makes much of a difference and not agree that it seems to be a bit awkward to place

Reply:
Yeah, I think it makes sense to use those if the type-guard can decide the type solely based on the value. Doesn't feel right to pass in `schemaProperty` to a typeguard like that. But indeed, I would probably use it for `isModelReference` for instance.